src/metabase/sync/sync_metadata/fks.clj

```diff
@@ -77,9 +77,6 @@
   [database :- i/DatabaseInstance]
   (reduce (fn [update-info table]
             (let [table-fk-info (sync-fks-for-table! database table)]
-              ;; Mark the table as done with its initial sync once this step is done even if it failed, because only
-              ;; sync-aborting errors should be surfaced to the UI (see [[sync-util/exception-classes-not-to-retry]]).
-              (sync-util/set-initial-table-sync-complete! table)
```

Comment:
`sync-fks!` is only called in `make-sync-steps`, so moving this to `make-sync-steps` is basically a refactor